### ISSUE TITLE
Leech Pipeline - FAILURE status handling

### ISSUE THREAD

**Issue by bradegler** [abcxyz/github-metrics-aggregator#58]
### TL;DR

Logs with a FAILURE status are currently filtered out of the query.

We need to figure out a way to mark how many attempts have been made for a particular delivery id preferably without causing an update to the row in BigQuery. 

The simplest approach would be to just not write a FAILURE status and let the query try again. 

The drawback is that something could be stuck forever in that situation and we wouldn't want to keep processing it. I think a secondary FAILURE table might work that we can join into the main query as WHERE count(failures where delivery_id = x) < 10 or something.

This adds complexity to the write operation though so it requires some thought.
